core/Enclave/enclave_hsm.cpp

```diff
@@ -43,767 +42,666 @@
+    {
+    case EH_RSA_2048:
+    case EH_RSA_3072:
+    case EH_RSA_4096:
+        ret = ehsm_rsa_verify(cmk,
+                              data,
+                              signature,
+                              result);
+        break;
+    // case EH_EC_P224:
+    case EH_EC_P256:
+        // case EH_EC_P384:
+        // case EH_EC_P512:
+        ret = ehsm_ecc_verify(cmk,
+                              data,
+                              signature,
+                              result);
+        break;
+    case EH_SM2:
+        ret = ehsm_sm2_verify(cmk,
+                              data,
+                              userid,
+                              signature,
+                              result);
+        break;
+    default:
+        printf("ecall verify unsupport keyspec.\n");
+        return SGX_ERROR_INVALID_PARAMETER;
     }
 
-    memset_s(&rsa_key, sizeof(rsa_params_t), 0, sizeof(rsa_params_t));
-    memset_s(&pri_key, sizeof(sgx_rsa3072_key_t), 0, sizeof(sgx_rsa3072_key_t));
-
     return ret;
 }
 
-sgx_status_t enclave_rsa_verify(const uint8_t *cmk_blob, size_t cmk_blob_size, const uint8_t *data, uint32_t data_len, const uint8_t *signature, uint32_t signature_len, bool* result)
+sgx_status_t enclave_generate_datakey(const ehsm_keyblob_t *cmk, size_t cmk_size,
+                                      const ehsm_data_t *aad, size_t aad_size,
+                                      ehsm_data_t *plaintext, size_t plaintext_size,
+                                      ehsm_data_t *ciphertext, size_t ciphertext_size)
 {
     sgx_status_t ret = SGX_ERROR_UNEXPECTED;
 
-    uint32_t sealed_rsa_len = sgx_calc_gcm_data_size(0, sizeof(rsa_params_t));
-    if (UINT32_MAX == sealed_rsa_len) {
-        printf("ecall rsa_verify failed to calculate sealed data size.\n");
-        return SGX_ERROR_UNEXPECTED;
+    if (cmk_size != APPEND_SIZE_TO_KEYBOB_T(cmk->keybloblen))
+    {
+        return SGX_ERROR_INVALID_PARAMETER;
     }
 
-    if (cmk_blob == NULL || cmk_blob_size < sealed_rsa_len + sizeof(sgx_rsa3072_public_key_t)) {
-        printf("ecall rsa_verify cmk_blob_size is too small.\n");
+    if (cmk->metadata.keyspec < 0)
+    {
         return SGX_ERROR_INVALID_PARAMETER;
     }
-
-    if (data == NULL || data_len == 0) {
-        printf("ecall rsa_verify data or data len is wrong.\n");
+    if (plaintext == NULL || ciphertext == NULL)
+    {
         return SGX_ERROR_INVALID_PARAMETER;
     }
-
-    if (result == NULL) {
-        printf("ecall rsa_verify result is NULL.\n");
+    if (plaintext->datalen > 1024 || plaintext->datalen == 0)
+    {
         return SGX_ERROR_INVALID_PARAMETER;
     }
-
-    if (signature == NULL || signature_len < RSA_OAEP_3072_SIGNATURE_SIZE) {
-        printf("ecall rsa_verify signature_len is too small than the expected 384.\n");
-        *result = false;
-        return SGX_SUCCESS;
+    if (cmk == NULL || cmk_size == NULL || cmk->keybloblen == NULL || cmk->keyblob == NULL)
+    {
```

Comment:
please refine the priority of the parameters check in each ecall interfaces.

Reply:
ok, all priority of the parameter will be check